src/main/java/goblintserver/GoblintServer.java

```diff
@@ -136,12 +122,10 @@ public StartedProcess runCommand(File dirPath, String[] command) throws IOExcept
 
             public void afterStop(Process process) {
                 if (process.exitValue() == 0) {
-                    magpieServer.forwardMessageToClient(new MessageParams(MessageType.Info, "Goblint server has stopped."));
                     log.info("Goblint server has stopped.");
                 } else if (process.exitValue() == 143) {
                     log.info("Goblint server has been killed.");
                 } else {
-                    magpieServer.forwardMessageToClient(new MessageParams(MessageType.Error, "Goblint server exited due to an error. Please check the output terminal of GobPie extension for more information."));
                     log.error("Goblint server exited due to an error. Please fix the issue reported above and restart the extension.");
                 }
             }
```

Comment:
When `Goblint` exits with an error, it always ends up in this method. This though did not for some reason display the pop-ups sent to `magpieServer`.  I tried to throw them further to `main`, where all the exceptions are now handled to notify the user through the pop-up there, but I cannot add a `throws` declaration for this method. 

Therefore, anytime when an error occurs due to `Goblint`'s running command being faulty or `Goblint` exiting due to another error, the user will not be notified and the error can only be seen from the terminal. So when the user does not have the output terminal open, they have no way of knowing that the extension is not working. This is pretty bad, but I couldn't figure out how to solve it.